langchain/load/serializable.py

```diff
@@ -0,0 +1,121 @@
+from abc import ABC
+from typing import Any, Dict, List, Literal, TypedDict, cast
+
+from pydantic import BaseModel, Field
+
+
+class BaseSerialized(TypedDict):
+    lc: int
+    id: List[str]
+
+
+class SerializedConstructor(BaseSerialized):
+    type: Literal["constructor"]
+    kwargs: Dict[str, Any]
+
+
+class SerializedSecret(BaseSerialized):
+    type: Literal["secret"]
+
+
+class SerializedNotImplemented(BaseSerialized):
+    type: Literal["not_implemented"]
+
+
+class Serializable(BaseModel, ABC):
+    @property
+    def lc_namespace(self) -> List[str]:
+        """
+        Return the namespace of the langchain object.
+        eg. ["langchain", "llms", "openai"]
+        """
+        return self.__class__.__module__.split(".")
+
+    @property
+    def lc_secrets(self) -> Dict[str, str]:
+        """
+        Return a map of constructor argument names to secret ids.
+        eg. {"openai_api_key": "OPENAI_API_KEY"}
+        """
+        return dict()
+
+    @property
+    def lc_attributes(self) -> List[str]:
```

Comment:
imo this should just be a dictionary - maximum flexibility (possible keys where attrs are saved dont line up with constructor args, eg if private

Reply:
So a dictionary of attribute name to constructor arg, yea?

Reply:
yeah so can just do `lc_kwargs.update(self.lc_attributes)`